pkg/kubelet/pod_manager.go

```diff
@@ -26,68 +26,84 @@ import (
 	"github.com/golang/glog"
 )
 
+// Pod manager stores and manages access to the pods.
+//
+// Kubelet discover pod updates from 3 sources: file, http, and apiserver.
+// Pods from non-apiserver sources are called static pods, and API server is
+// not aware of the existence of static pods. In order to monitor the status of
+// such pods, kubelet create a mirror pod for each static pod via the API
+// server.
+//
+// A mirror pod has the same pod full name (name and namespace) as its static
+// counterpart (albeit different metadata such as UID, etc). By leveraging the
+// fact that kubelet reports the pod status using the pod full name, the status
+// of the mirror pod always reflects the acutal status of the static pod.
+// When a static pod gets deleted, the associated orphaned mirror pods will
+// also be removed.
+
 type podManager interface {
-	UpdatePods(u PodUpdate, podSyncTypes map[types.UID]metrics.SyncPodType)
-	GetPods() ([]api.Pod, mirrorPods)
-	GetPodByName(namespace, name string) (*api.Pod, bool)
+	GetPods() []api.Pod
 	GetPodByFullName(podFullName string) (*api.Pod, bool)
-	TranslatePodUID(uid types.UID) types.UID
-	DeleteOrphanedMirrorPods(mirrorPods *mirrorPods)
+	GetPodByName(namespace, name string) (*api.Pod, bool)
+	GetPodsAndMirrorMap() ([]api.Pod, map[string]*api.Pod)
 	SetPods(pods []api.Pod)
-	mirrorManager
+	UpdatePods(u PodUpdate, podSyncTypes map[types.UID]metrics.SyncPodType)
+	DeleteOrphanedMirrorPods()
+	TranslatePodUID(uid types.UID) types.UID
+	mirrorClient
 }
 
+// All arrays/maps in basicPodManager should be set by calling UpdatePods();
+// individual arrays/maps are not immutable and no other methods should attempt
+// to modify them.
 type basicPodManager struct {
-	// Protects all internal pod storage/mappings.
+	// Protects all internal array/maps.
 	lock sync.RWMutex
+	// Regular pods (including both static and non-static pods).
```

Comment:
I'm not sure I understand your suggestion. The arrays are used to store the content of the pods. Are you saying that we keep the pods in the map directly (i.e. `map[types.UID]api.Pod`?). Since we need to have two maps (keyed by UID and full name individually), we will save one as api.Pod and the other as *api.Pod?